### PR TITLE
Add ability to encrypt file to be shared and fix ticket serialization

### PR DESCRIPTION
**Pull request by andrii-iakovenko**

### CODE REVIEW

**Review by SergeySeroshtan — DISMISSED** [review]

**Review by SergeySeroshtan — DISMISSED** [review]

File: ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt [hunk 1]
```diff
@@ -119,6 +139,106 @@ class PeerToPeerTest {
...
+                            return TestUtils.generateTokenString(identityTwo)
+                        }
+                    },
+                    TestConfig.context,
+                    enableRatchet = false,
+                    keyRotationInterval = TimeSpan.fromTime(3600, TimeUnit.SECONDS))
+            ethreeTwo.register().execute()
+            ethrees[identityTwo] = ethreeTwo
+
+//            cards[identityTwo] = ethree.findUser(identityTwo).get()
```
Comment (SergeySeroshtan) [inline ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt]: [CodeStyle]: Remove commented code.

File: ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt [hunk 2]
```diff
@@ -119,6 +139,106 @@ class PeerToPeerTest {
...
+            encryptedMessages.add(Pair(identityTwo, encryptedMessage))
+        }
+
+        // Decrypt many messages at the same time
+        val decryptedMessages = ConcurrentLinkedQueue<String>()
+        runBlocking {
+            encryptedMessages.forEach { pair ->
+                launch(Dispatchers.IO) {
+//                    Log.d(TAG, "Start decryption for ${pair.first}")
+//                    decryptedMessages.add(ethree.authDecrypt(pair.second, cards[pair.first]))
```
Comment (SergeySeroshtan) [inline ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt]: [CodeStyle]: Remove commented code.

File: ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt [hunk 3]
```diff
@@ -119,6 +139,106 @@ class PeerToPeerTest {
...
+                    decryptedMessages.add(ethree.authDecrypt(pair.second, ethree.findUser(pair.first).get()))
+//                    Log.d(TAG, "Finished decryption for ${pair.first}")
```
Comment (SergeySeroshtan) [inline ethree-common/src/androidTest/java/com/virgilsecurity/android/common/worker/PeerToPeerTest.kt]: [CodeStyle]: Remove commented code.

**Review by SergeySeroshtan — CHANGES_REQUESTED** [review]

File: ethree-common/src/main/java/com/virgilsecurity/android/common/worker/StreamsEncryptWorker.kt [hunk 4]
```diff
@@ -94,6 +95,27 @@ internal class StreamsEncryptWorker internal constructor(
         return decryptInternal(inputStream, outputStream, card.publicKey)
     }
 
+    internal fun encryptShared(inputStream: InputStream,
+                               inputStreamSize: Int,
+                             outputStream: OutputStream): ByteArray {
+        if (inputStream.available() == 0) throw EmptyArgumentException("inputStream")
```
Comment (SergeySeroshtan) [inline ethree-common/src/main/java/com/virgilsecurity/android/common/worker/StreamsEncryptWorker.kt]: Empty streams are allowed for encryption

**Review by SergeySeroshtan — CHANGES_REQUESTED** [review]

File: ethree-common/src/main/java/com/virgilsecurity/android/common/worker/StreamsEncryptWorker.kt (hunk 4 above)
Comment (andrii-iakovenko) [reply ethree-common/src/main/java/com/virgilsecurity/android/common/worker/StreamsEncryptWorker.kt]: Fixed

**Review by SergeySeroshtan — APPROVED** [review]

**Pull request merged**